src/Features/CSharp/Portable/ConvertToRecord/ConvertToRecordCommon.cs

```diff
@@ -339,14 +370,23 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
                     .WithLeadingTrivia(modifiedClassTrivia)
                     .WithAdditionalAnnotations(Formatter.Annotation);
 
-            var changedDocument = await document.ReplaceNodeAsync(
-                typeDeclaration, changedTypeDeclaration, cancellationToken).ConfigureAwait(false);
+            var solutionEditor = new SolutionEditor(document.Project.Solution);
+            var currDocEditor = await solutionEditor.GetDocumentEditorAsync(document.Id, cancellationToken).ConfigureAwait(false);
+            currDocEditor.ReplaceNode(typeDeclaration, changedTypeDeclaration);
 
-            return changedDocument;
+            await RefactorInitializersAsync(type, solutionEditor, positionalParamSymbols, cancellationToken).ConfigureAwait(false);
+            return solutionEditor.GetChangedSolution();
         }
 
-        private static SyntaxList<AttributeListSyntax> GetModifiedAttributeListsForProperty(PropertyDeclarationSyntax p)
-            => SyntaxFactory.List(p.AttributeLists.SelectAsArray(attributeList =>
+        private static SyntaxList<AttributeListSyntax> GetModifiedAttributeListsForProperty(PositionalParameterInfo result)
+        {
+            if (result.IsInherited || result.KeepAsOverride)
+            {
+                // for both of these, the attributes on the other definition track so we don't need to redeclare
```

Comment:
i don't know you mean by "the attributes on the other definition track".  can you give an example?